angularSTEP/src/app/app-routing.module.ts

```diff
@@ -14,8 +15,10 @@ const routes: Routes = [{ path: '', redirectTo: 'home', pathMatch: 'full' },
                         { path: 'signup', component: SignupComponent },
                         { path: 'login-email', component: EmailComponent },
                         { path: 'home', component: HomeComponent },
+                        { path: 'users', component: ViewProfilesComponent},
+                        { path:'users/:username', component:UserPageComponent },
                         { path: 'upload-recipe', component: UploadRecipeComponent},
-                        { path: 'confirm-upload', component: ConfirmComponent},];
+                        { path: 'confirm-upload', component: ConfirmComponent}];
```

Comment:
Missing a space between `path:` and `'users/:username'` as well as between `component:` and `UserPageComponent`

Missing a space before some of the closing braces.